pywrap.bzl

```diff
@@ -165,6 +172,79 @@ def pybind_extension(
         deps = ["%s" % cc_library_name],
     )
 
+def _generated_pywrap_win_def_file_impl(ctx):
+    cc_toolchain = find_cpp_toolchain(ctx)
+    feature_configuration = cc_common.configure_features(
+        ctx = ctx,
+        cc_toolchain = cc_toolchain,
+        requested_features = ctx.features,
+        unsupported_features = ctx.disabled_features,
+    )
+
+    dependency_libraries = []
+
+    pywrap_index = ctx.attr.pywrap_index
+    pywrap_infos = ctx.attr.dep[CollectedPywrapInfo].pywrap_infos.to_list()
+    if pywrap_index >= 0:
+        pywrap_infos = [pywrap_infos[pywrap_index]]
+
+    for pywrap_info in pywrap_infos:
+        cc_info = pywrap_info.cc_info
+
+        linker_inputs = cc_info.linking_context.linker_inputs.to_list()
+        
+        for linker_input in linker_inputs:
+            for lib in linker_input.libraries:
+                lib_copy = lib;
+                if not lib.alwayslink:
+                    lib_copy = cc_common.create_library_to_link(
+                        actions = ctx.actions,
+                        cc_toolchain = cc_toolchain,
+                        feature_configuration = feature_configuration,
+                        static_library = lib.static_library,
+                        pic_static_library = lib.pic_static_library,
+                        interface_library = lib.interface_library,
+                        alwayslink = True,
+                    )
+                if lib_copy.objects:
+                    dependency_libraries.append(lib_copy)
+
+    pywrap_infos = ctx.attr.dep[CollectedPywrapInfo].pywrap_infos.to_list()
+    pywrap_info = pywrap_infos[ctx.attr.pywrap_index]
+    win_def_file_name = "%s.gen.def" % pywrap_info.owner.name
+    win_def_file = ctx.actions.declare_file(win_def_file_name)
+    print("win_def_file.path:", win_def_file.path)
+    print("dependency_libraries[0][static_library].path:", dependency_libraries[0].static_library.path)
+    ctx.actions.run_shell(
+        inputs = [dependency_library.static_library for dependency_library in dependency_libraries],
+        # command = "echo \"EXPORTS\r\n\t?first_func@@YAHH@Z\r\n\t?second_func@@YAHH@Z\r\n\">> {win_def_file}".format(
+        command = "echo \"EXPORTS\r\n\">> {win_def_file} && \"C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.29.30133\\bin\\Hostx86\\x86\\dumpbin.exe\" //SYMBOLS {dependency_library} | grep 'UNDEF' | grep 'External' | grep '()' | cut -d\" \" -f16 | grep '\\?[a-zA-Z]\\+.*'>> {win_def_file}".format(
```

Comment:
We cannot harcode paths to dumpbin.exe (or any other tool) like that. This will not work on any machine which installed MSVC in a different locaiton, plus it will break on any other version (and this one hardcodes even the incremental/patch version of it). Also it very non-hermetic, and would require changing rule itself in case of version upgrade.